include/swift/AST/Decl.h

```diff
@@ -819,7 +819,11 @@ class alignas(1 << DeclAlignInBits) Decl {
 
   // Make vanilla new/delete illegal for Decls.
   void *operator new(size_t Bytes) = delete;
+
+  // Work around MSVC error: attempting to reference a deleted function.
```

Comment:
> Ah, maybe you would then have to delete operator delete in all the subclasses too.

This doesn't work - it just changes the error message from
> void swift::Decl::operator delete(void *)': attempting to reference a deleted function

to 
>void swift::ModuleDecl::operator delete(void *)': attempting to reference a deleted function

Making it virtual causes the error

> 'delete' cannot be a virtual function